sh/configure_env.sh

```diff
@@ -0,0 +1,112 @@
+#
+# Author: Philip Deegan
+# Email : philip.deegan@polytechnique.edu 
+# Date  : 21 - September - 2017
+#
+# This script is to be included to setup variables to build tick
+#   It is done so to be cross platform and heterogeneous
+#
+# Variables
+#  $PY = "python" binary
+#  $PCONF = "python-config" binary
+#  $LDARGS=Linking arguments
+#  $LIB_POSTFIX= python library format 
+#  $LIB_POSTEXT= system library file extention
+#  $PROFILES= array of mkn profiles to build
+# 
+######################################################################
+
+set -e
+
+# Finding the system python binary
+[ -z "$PY" ] && which python3 &> /dev/null && PY="python3"
+[ -z "$PY" ] && which python &> /dev/null && PY="python"
+[ -z "$PY" ] && echo "python or python3 not on path, exiting with error" && exit 1
```

Comment:
We should check that python 3 is used. 
With python 2 I get the following, not so informative error
```
tick/base/array/src/alloc.h:20:3: error: use of undeclared identifier 'PyMem_RawFree'; did you mean 'PyMem_Free'?
  PyMem_RawFree(reinterpret_cast<void*>(ptr));
  ^~~~~~~~~~~~~
  PyMem_Free
```
Do we check anywhere that user has SWIG installed ?

Reply:
python3 is attempted to be used by default - it can be overridden by executing

```
export PY=/path/to/python3
```

Similarly for PCONF

```
export PCONF=/path/to/python-config
```

in "mkn.sh" we attempt to use swig. so it will fail if it not found - swig is mentioned in the setup requirements in the README

Reply:
I've added checks for

- to see if swig is on path (overridable variable $SWIG)
- that python found is version 3